app/src/main/java/com/example/androidlearning/MyNotifications.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.androidlearning
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+
+
+class MyNotifications(private val context: Context) {
+
+   private val notificationManagerCompact = NotificationManagerCompat.from(context)
+    private val channelId = "my_channel_id"
+
+
+    fun headsUpNotification(){
+        val notificationId = 1
+        val intent = Intent(context, PressNotification::class.java).apply {
+            flags = Intent.FLAG_ACTIVITY_NEW_TASK
+        }
+        val pendingIntent: PendingIntent = PendingIntent.getActivity(context, notificationId, intent, PendingIntent.FLAG_IMMUTABLE)
```

Comment:
What does FLAG_IMMUTABLE means? and can you please let me know other available flags and its purpose?

Reply:
FLAG_IMMUTABLE - configuration cannot be changed after creation, used for better security 
FLAG_UPDATE_CURRENT - if the described PendingIntent already exists, then keep it but replace its extra data with what is in this new Intent
FLAG_ONE_SHOT - can be used only once.
FLAG_NO_CREATE: If the PendingIntent doesn't already exist, don't create it, returns null. This is useful for checking whether a PendingIntent with a given configuration already exists.
FLAG_CANCEL_CURRENT: If the PendingIntent already exists, it will be canceled and replaced with a new one.This is useful when you want to ensure that only the latest intent is delivered.
FLAG_MUTABLE: This flag indicates that the PendingIntent is mutable.